src/main/java/org/broadinstitute/hellbender/utils/bwa/BwaMemIndex.java

```diff
@@ -0,0 +1,349 @@
+package org.broadinstitute.hellbender.utils.bwa;
+
+import htsjdk.samtools.Cigar;
+import htsjdk.samtools.CigarElement;
+import htsjdk.samtools.CigarOperator;
+import htsjdk.samtools.SAMFlag;
+import org.apache.spark.api.java.JavaSparkContext;
+import org.broadinstitute.hellbender.exceptions.GATKException;
+import org.broadinstitute.hellbender.exceptions.UserException;
+import org.broadinstitute.hellbender.utils.NativeUtils;
+
+import java.io.FileInputStream;
+import java.io.IOException;
+import java.nio.ByteBuffer;
+import java.nio.ByteOrder;
+import java.util.ArrayList;
+import java.util.List;
+
+public final class BwaMemIndex implements AutoCloseable {
+    private long indexAddress;
+    private final List<String> refContigNames;
+    private static volatile BwaMemIndex globalInstance;
+
+    public static void createIndexImage( final String refName, final String imgName ) {
+        assertNonEmptyReadable(refName+".amb");
+        assertNonEmptyReadable(refName+".ann");
+        assertNonEmptyReadable(refName+".bwt");
+        assertNonEmptyReadable(refName+".pac");
+        assertNonEmptyReadable(refName+".sa");
+        createIndexImageFile(refName, imgName);
+    }
+
+    public static BwaMemIndex getInstance( final String indexImageFile ) {
+        BwaMemIndex result = globalInstance;
+        if ( result == null ) {
+            synchronized(BwaMemIndex.class) {
+                result = globalInstance;
+                if ( result == null ) globalInstance = result = new BwaMemIndex(indexImageFile);
+            }
+        }
+        return result;
+    }
+
+    public static void closeInstance() {
+        BwaMemIndex result = globalInstance;
+        if ( result != null ) {
+            synchronized(BwaMemIndex.class) {
+                result = globalInstance;
+                if ( result != null ) {
+                    globalInstance = null;
+                    result.close();
+                }
+            }
+        }
+    }
+
+    public static void closeAllDistributedInstances( final JavaSparkContext ctx ) {
+        int nJobs = ctx.defaultParallelism();
```

Comment:
I couldn't find a much better solution, but you could use the number of partitions instead of number of workers. 

I thought maybe Spark wouldn't distribute this evenly to executors evenly in the case there are still tasks being performed. Also is it possible, based on what I [read](https://spark.apache.org/docs/latest/job-scheduling.html#remove-policy) here, that executors can be decommissioned based on timeout, meaning some might get shut down without closing it properly? Not sure if that would ever be the case here.